waku/v2/protocol/store/waku_store.go

```diff
@@ -374,31 +374,35 @@ func (store *WakuStore) storeIncomingMessages(ctx context.Context) {
 
 func (store *WakuStore) onRequest(s network.Stream) {
 	defer s.Close()
-
+	logger := store.log.With(logging.HostID("peer", s.Conn().RemotePeer()))
 	historyRPCRequest := &pb.HistoryRPC{}
 
 	writer := protoio.NewDelimitedWriter(s)
 	reader := protoio.NewDelimitedReader(s, math.MaxInt32)
 
 	err := reader.ReadMsg(historyRPCRequest)
 	if err != nil {
-		store.log.Error("error reading request", err)
+		logger.Error("reading request", zap.Error(err))
 		metrics.RecordStoreError(store.ctx, "decodeRPCFailure")
 		return
 	}
-
-	store.log.Info(fmt.Sprintf("%s: Received query from %s", s.Conn().LocalPeer(), s.Conn().RemotePeer()))
+	logger = logger.With(zap.String("id", historyRPCRequest.RequestId))
```

Comment:
This is the one that I embellished a bit with some additional parameters that we found useful in our fork. It at least shows how to use the history query field helpers. Happy to rip it out if you don't find it useful, but figured I can at least show how it was meant to be used.